src/d3dx12_property_format_table.cpp

```diff
@@ -1465,14 +1468,18 @@ D3D_FORMAT_LAYOUT D3D12_PROPERTY_LAYOUT_FORMAT_TABLE::GetLayout(DXGI_FORMAT Form
 // GetComponentName
 D3D_FORMAT_COMPONENT_NAME D3D12_PROPERTY_LAYOUT_FORMAT_TABLE::GetComponentName(DXGI_FORMAT Format, UINT AbsoluteComponentIndex)
 {
-    D3D_FORMAT_COMPONENT_NAME name;
+    D3D_FORMAT_COMPONENT_NAME name = {};
     switch( AbsoluteComponentIndex )
     {
     case 0: name = s_FormatDetail[GetDetailTableIndexNoThrow( Format )].ComponentName0; break;
     case 1: name = s_FormatDetail[GetDetailTableIndexNoThrow( Format )].ComponentName1; break;
     case 2: name = s_FormatDetail[GetDetailTableIndexNoThrow( Format )].ComponentName2; break;
     case 3: name = s_FormatDetail[GetDetailTableIndexNoThrow( Format )].ComponentName3; break;
-    default: throw E_FAIL;
+#if defined(__cpp_exceptions) && (!defined(_HAS_EXCEPTIONS) || _HAS_EXCEPTIONS != 0)
+    default: throw std::invalid_argument("AbsoluteComponentIndex");
```

Comment:
D3D's convention is typically to use `_com_error`, so this would be `throw _com_error(E_FAIL);`

Reply:
``_com_error`` is highly problematic and non-portable. Since this file is `.w` processed internally, can we make it use ``_com_error`` for internal scenarios but stick with ``stdexcept`` for public ones?

Reply:
I'll leave this PR pending until I work that out and submit it to Direct3D

Reply:
The internal source-of-truth here now generates this version of the headers, but leaves the older behavior without disturbing internal tests.